### ISSUE TITLE
header-cell.component: AoT compilation problem

### ISSUE THREAD

**Issue by Kopleman** [swimlane/ngx-datatable#389]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
AoT compilation throws error 'header-cell.component.ngfactory.ts:287:113: Operator '===' cannot be applied to types 'SelectionType' and 'string''

**Expected behavior**
No errors

**Reproduction of the problem**

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Table version:** 0.7.x
4.0.0

* **Angular version:** 2.0.x
2.4.0
* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
TypeScript 2.0.10

**Comment by j3ddesign** [comment]
I'm having the same issue when building for prod.
The repo this is occurring on for me currently is https://github.com/j3ddesign/uems-website incase you need easy reproduction.

The error occurs when running `npm run compile`

**Comment by amcdnl** [comment]
Fixed in 4.1
